Pull Request: Updates packager with new version supports card as application type.

tohman opened the pull request:
Removed validation for application types (_application_ and _viewer_) and filter attributes (_exts_ and _uris_).
Also it has Framework submodule with corresponding changes.

More info [here](https://github.com/blackberry-webworks/WebWorks-API-Docs/pull/138)
User story mentions this issue [link](https://github.com/blackberry/BB10-Webworks-Packager/issues/112) but no card are mentioned there.

[JSDoc](http://blackberry-webworks.github.com/WebWorks-API-Docs/WebWorks-API-Docs-next-BB10-cards/view/index.html). 
[CI Build](http://ci0000003863287:8080/hudson/job/BB10-Webworks-Packager-next-adds-card-invocation-support/4/)

Commit:
Updates packager with new versoin support card, update Framework submodule to support cards

cdelcol commented:
Typo in title of pull request :)

tohman commented:
Revised - typo eliminated.

ishneur commented:
@nukulb The docs for blackberry.invoke card events are wrong - http://blackberry-webworks.github.com/WebWorks-API-Docs/WebWorks-API-Docs-next-BB10-cards/view/blackberry.invoke.html

The event names in the docs are onChildCardClosed, onChildCardEndPeek and onChildCardStartPeek, but their actual names are onCardChildClosed, onCardEndPeek and onCardStartPeek, which is why I couldn't get those events to work in my app.

ishneur commented:
@nukulb Another problem is that the doc says that the callback function for onCardEndPeek takes an argument (peekType), but when I display the value of that argument it's "undefined".

In Sergey's app it doesn't take an argument. So it's either not supposed to take an argument, or it's a bug where it's supposed to have a value just like the doc says ("content" or "root"), or it's just not supposed to be there in the callback function.

nukulb commented:
@tohman should be able to fix this up.

------Original Message------
From: Igor Shneur
To: Nukul Bhasin
Subject: Re: [BB10-Webworks-Packager] Pull Request: Updates packager with new version supports card as application type. (#71)
Sent: Jul 27, 2012 4:33 PM

@nukulb Another problem is that the doc says that the callback function for onCardEndPeek takes an argument (peekType), but when I display the value of that argument it's "undefined".

In Sergey's app it doesn't take an argument. So it's either not supposed to take an argument, or it's a bug where it's supposed to have a value just like the doc says ("content" or "root"), or it's just not supposed to be there in the callback function.

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/71#issuecomment-7315490

Sent from my BlackBerry device on the Rogers Wireless Network

ishneur commented:
Found those issues:
1. I cannot invoke a card from an opened card:
   From the parent app, I open the first card - InvokedCard01. From InvokedCard01, I'm trying to open another card called InvokedCard02, which doesn't work. If I just open InvokedCard01 as an app (not as a card), then I can invoke the card InvokedCard02 from it. All of those are webworks apps.
2. The peekType in the onCardStartPeek callback in the parent is always "content", even if I call blackberry.invoked.cardStartPeek('root') from the child card.

ishneur commented:
I noticed another issue: The onSuccess callback of blackberry.invoke.invoke is not being fired in the parent the very first time a card is invoked, but if you close the card and invoke it again, then it is fired.

You can use those apps to reproduce it:
https://github.com/downloads/blackberry-webworks/BB10-Webworks-Packager/InvokeCardAPI.zip
https://github.com/downloads/blackberry-webworks/BB10-Webworks-Packager/InvokedCard01.zip

Start InvokeCardAPI and click on "Bound - Invoke a WebWorks card - InvokedCard01", then click "requestClosure()" in the invoked card to go back to the parent.

ishneur commented:
@tohman Can "blackberry.invoked.cardResizeDone()" be actually tested? When I call it, nothing happens. Does it actually do anything?

According to the docs, it's - "As a Card, notify the navigator that it is finished handling resize request".

http://blackberry-webworks.github.com/WebWorks-API-Docs/WebWorks-API-Docs-next-BB10-cards/view/blackberry.invoked.html#.cardResizeDone

tohman commented:
I don't see how it can be tested there is no callback or return param for this method.

tohman commented:
I've updated doc for onCardEndPeek now the callback doesn't mention the peekType parameter also events name now updated for invoker, also i was able to invoke card from another invoked card in the trunk build.
Links to CI and Doc have been updated.

ishneur commented:
I tested Sergey's last fixes in trunk 1741 and they work correctly now. I'm also able to invoke a second card from an opened card now.

Here is a summary of outstanding issues:
1. startPeek('root') doesn't work - it always peeks to the content (immediate parent) and not the root, and the callback value for event onChildCardStartPeek is also always "content"
2. The onSuccess callback of blackberry.invoke.invoke is not being fired in the parent the first time a card is invoked. It only happens on a fresh OS install or after device reboot. After that, if you reopen the same card, it does fire.
3. Event listeners stay running in cards even after the card is closed and after its parent app is closed. I also noticed that if I change the card's code/HTML and reinstall it, the old content is still shown until I reboot the device and reinstall the card.
4. I cannot get the onCardClosed event to fire when a card is closed so I might not be able to test it.

Commit:
Updates Framework submodule

tohman commented:
Merged manually
